### PR TITLE
fix(query): fix pageParam for useInfiniteQuery

### LINKED ISSUE #765: useInfiniteQuery: pageParams are overridden in queryFn when trying to implement infinite scroll functionality.
Might be my specific use case, but I can't seem to increment the pageParam in my infinite query. It gets overidden by the rest operator in the queryFn.

## What are the steps to reproduce this issue?

Relevant orval config  
```typescript     
operations: {
  get_all_utterances_for_conversation_paginated: {
      query: {
        useQuery: true,
        useInfinite: true,
        useInfiniteQueryParam: "page", // <----- I want to increment this
      },
    },
  },
```
Generated queryFn
```typescript
  const queryFn: QueryFunction<Awaited<ReturnType<typeof getAllUtterancesForConversationPaginated>>> = ({ pageParam }) => getAllUtterancesForConversationPaginated(orgUuid,conversationUuid,{ page: pageParam, ...params }, requestOptions);
```

Request
```typescript
export const getAllUtterancesForConversationPaginated = (
    orgUuid: string,
    conversationUuid: string,
    params?: GetAllUtterancesForConversationPaginatedParams,
    options?: SecondParameter<typeof customInstance>,) =>  {
    console.log(params?.page) // <------- this doesn't  increment. 
      return customInstance<UtteranceReadPaginated>(
      {url: `/api/org/${orgUuid}/conversation/${conversationUuid}/utterances-paginated`, method: 'get',
        params 
    },
      options);
    }
```

## What happens?

The pageParam does not update as expected because it is overridden by the rest operator. 

```typescript
  const queryFn: QueryFunction<Awaited<ReturnType<typeof getAllUtterancesForConversationPaginated>>> = ({ pageParam }) => getAllUtterancesForConversationPaginated(orgUuid,conversationUuid,{ page: pageParam, ...params }, requestOptions);  // <------ HERE
```
…

## What were you expecting to happen?
I think the `...params` rest operator should go before the pageParam, like this:

```typescript
{ ...params, page: pageParam,}
```
This way we can update the page param, and spread all other params into the request

…

## Any logs, error output, etc?

…

## Any other comments?

…

## What versions are you using?

**Operating System:** …
**Package Version:** …
**Browser Version:** …

### PR DESCRIPTION
**Pull request by georgiev-anton**
## Status

**READY**

## Description

Small fix for **useInfiniteQuery**

Without this fix, page switching with the getNextPageParam function does not work

Fix #765

### CODE REVIEW

File: packages/query/src/index.ts [hunk 1]
```diff
@@ -731,7 +731,7 @@ const generateQueryImplementation = ({
           )
             return param.destructured;
           return param.name === 'params'
-            ? `{ ${queryParam}: pageParam, ...params }`
+            ? `{...params, ${queryParam}: pageParam || params.page }`
```
Comment (anymaniax) [inline packages/query/src/index.ts]: Hello @georgiev-anton and thanks for the pr.  Why params.page here?

Comment (georgiev-anton) [reply packages/query/src/index.ts]: Hello, because without this page switching does not work

Comment (anymaniax) [reply packages/query/src/index.ts]: Here you force everyone to pass a page params. I use infinite query a lot without the need of this

**Review by anderson-comvex — APPROVED** [review]

**Review by comvex-taisuke-j — APPROVED** [review]

File: packages/query/src/index.ts (hunk 1 above)
Comment (comvex-taisuke-j) [reply packages/query/src/index.ts]: I think we need this if the initial query requires the page parameter
https://github.com/anymaniax/orval/issues/765

**Comment by melloware** [comment]
Is there an open ticket that I can link this PR to?  I see 4 open infinite query tickets not sure if this PR resolves any of them?

**Comment by georgiev-anton** [comment]
> Is there an open ticket that I can link this PR to? I see 4 open infinite query tickets not sure if this PR resolves any of them?

yes this https://github.com/anymaniax/orval/issues/765

**Comment by anderson-comvex** [comment]
This PR fix an issue where the initial page params needs to be specified and needs to keep being incremented via the nextPage fetcher.

**Review by melloware — APPROVED** [review]

**Comment by melloware** [comment]
@georgiev-anton looks like tests are failing here too?

**Comment by georgiev-anton** [comment]
> @georgiev-anton looks like tests are failing here too?

Yes, but this one requires more work. 

`params.page` parameter can be undefined, it must come from the request function or from the server response

**Comment by melloware** [comment]
Yeah this is tough because we won't know at build time will we?

Is it possible for every infinite query to generate two queries one the way it works now and a second one that has the params the way your need it?

File: packages/query/src/index.ts (hunk 1 above)
Comment (anymaniax) [reply packages/query/src/index.ts]: v5 fix this no? With `initialPageParam `?

Comment (anymaniax) [reply packages/query/src/index.ts]: and for v4 it should be `{...params, ${queryParam}: pageParam || params?.['${queryParam}']}` if you want it to work no?

**Comment by melloware** [comment]
@georgiev-anton did you see @anymaniax feedback?  I would love to get this fixed as it closes about 4 open tickets.

**Comment by georgiev-anton** [comment]
> @georgiev-anton did you see @anymaniax feedback? I would love to get this fixed as it closes about 4 open tickets.

yes  @anymaniax thanks it works

**Comment by melloware** [comment]
Nice but looks like a test failure now?

**Comment by georgiev-anton** [comment]
> Nice but looks like a test failure now?

yes I am correcting

**Comment by georgiev-anton** [comment]
@melloware ready, please check

**Review by melloware — APPROVED** [review]

**Comment by melloware** [comment]
@georgiev-anton the only remaining comment from @anymaniax i don't understand is 

> v5 fix this no? With `initialPageParam` ?

Is this resolved too?

**Comment by georgiev-anton** [comment]
> @georgiev-anton the only remaining comment from @anymaniax i don't understand is
> 
> > v5 fix this no? With `initialPageParam` ?
> 
> Is this resolved too?

Yes

**Comment by georgiev-anton** [comment]
initialPageParam bug resolved

**Comment by melloware** [comment]
Nice great job.  I sent @anymaniax to get his eyes on it one last time before I merge it!

**Review by anymaniax — APPROVED** [review]

**Pull request merged**